sdk/keyvault/Azure.Security.KeyVault.Keys/src/RecoverDeletedKeyOperation.cs

```diff
@@ -5,34 +5,37 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Azure.Core;
-using Azure.Core.Pipeline;
 
 namespace Azure.Security.KeyVault.Keys
 {
     /// <summary>
     /// A long-running operation for <see cref="KeyClient.StartRecoverDeletedKey(string, CancellationToken)"/> or <see cref="KeyClient.StartRecoverDeletedKeyAsync(string, CancellationToken)"/>.
     /// </summary>
-    public class RecoverDeletedKeyOperation : Operation<KeyVaultKey>
+    public class RecoverDeletedKeyOperation : Operation<KeyVaultKey>, IOperation<KeyVaultKey>
```

Comment:
Could we just have `Operation<T>` implement `IOperation<T>`, or do we explicitly not include that source in Core? Could/should it be to save steps/complexity?

Reply:
We would need to add a new abstract method to `Operation<T>` if we did that, which would be breaking. We could add it as a virtual method and provide default implementations to the LRO members, and expose `OperationState<T>`, but I don't think it's worth the risk of making big changes in Core. We would only gain:
- We wouldn't need to add `IOperation<T>` to the concrete LRO class declarations.
- We wouldn't need to import `OperationInternal` in the project file.
- We wouldn't need to instantiate an `OperationInternal` object internally.